terra_layer/style.py

```diff
@@ -0,0 +1,397 @@
+            break
+        last_value_id -= 1
+
+    # loop over all values that are large enough
+    for limit_id in range(1, last_value_id + 1):
+        v = values[limit_id - 1]
+        # compute the height of the symbol
+        h = circle_boundaries_value_to_symbol_height(v, max_value, max_size)
+        # do not draw the symbol if it is too close to the smallest symbol (but is not the smallest limit itself)
+        if h - last_height < dmin and limit_id != last_value_id:
+            continue
+        # do not draw the symbol if it is too close to the previously drawn symbol
+        if previous_height - h < dmin:
+            continue
+        filtered_values.append(v)
+        # remember the height of the last drawn symbol
+        previous_height = h
+
+    return filtered_values
+
+
+def gen_legend_circle(min, max, size):
+    """
+    Generate a circle legend.
+    """
+    candidates = circle_boundaries_candidate(min, max)
+    candidates = [max] + candidates + [min]
+    boundaries = circle_boundaries_filter_values(candidates, min, max, size / 20)[::-1]
+    return [
+        {"radius": b * size / max, "label": f"{b}", "shape": "circle"}
+        for b in boundaries
+    ]
+
+
+def gen_layer_fill(fill_color=DEFAULT_FILL_COLOR, stroke_color=DEFAULT_STROKE_COLOR):
+    """
+    Build a Mapbox GL Style layer for pylygon fill.
+    """
+    return {
+        "type": "fill",
+        "paint": {"fill-color": fill_color, "fill-outline-color": stroke_color},
+    }
+
+
+def gen_layer_circle(
+    radius, fill_color=DEFAULT_FILL_COLOR, stroke_color=DEFAULT_STROKE_COLOR
+):
+    """
+    Build a Mapbox GL Style layer for circle.
+    """
+    return {
+        "type": "circle",
+        "paint": {"circle-radius": radius, "circle-stroke-color": stroke_color},
+    }
+
+
+def generate_style_from_wizard(layer, config):
+    """
+    Return a Mapbox GL Style and a Legend from a wizard setting.
+    """
+    geo_layer = layer.source.get_layer()
+    symbology = config["symbology"]
+    property = config["property"]
+
+    if symbology == "graduated":
+        # {
+        #     "property": "my_property",
+        #     "symbology": "graduated",
+        #     "method": "equal_interval",
+        #     "fill_color": ["#ff0000", "#aa0000", "#770000", "#330000", "#000000"],
+        #     "stroke_color": "#ffffff",
+        # }
+        colors = config["fill_color"]
+        boundaries = discretize(geo_layer, property, config["method"], len(colors))
+        if boundaries:
+            style = gen_layer_fill(
+                fill_color=gen_style_steps(
+                    get_property_style(property), boundaries, colors
+                ),
+                stroke_color=config.get("stroke_color", DEFAULT_STROKE_COLOR),
+            )
+            legend = gen_legend_steps(boundaries, colors)
+            return {"style": style, "legend": legend}
+        else:
+            return {"style": {}, "legend": []}
+
+    elif symbology == "circle":
```

Comment:
And replace circle by "bubble" ? https://python-graph-gallery.com/bubble-map/

Reply:
I can rename it "Proportional symbol" with a sub option "circle", but bubble is childish.

http://wiki.gis.com/wiki/index.php/Proportional_symbol_map